experimental/saiexperimentaldashtag.h

```diff
@@ -0,0 +1,367 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiexperimentaldashtag.h
+ *
+ * @brief   This module defines SAI P4 extension  interface
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHTAG_H_)
+#define __SAIEXPERIMENTALDASHTAG_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASH_TAG SAI - Extension specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Entry for dst_tag_entry
+ */
+typedef struct _sai_dst_tag_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief LPM matched key dip
+     */
+    sai_ip_prefix_t dip;
+
+} sai_dst_tag_entry_t;
+
+/**
+ * @brief Attribute ID for dash_tag_dst_tag_entry
+ */
+typedef enum _sai_dst_tag_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_DST_TAG_ENTRY_ATTR_START,
+
+    /**
+     * @brief Action set_dst_tag parameter TAG_MAP
+     *
+     * @type sai_uint32_t
```

Comment:
DASH behavioral model needs to be changed to make this attribute's type sai_object_list_t, per discussion with @prsunny